napari/utils/_tests/test_progress.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+import pytest
+
+try:
+    import qtpy  # noqa
+except ImportError:
+    pytest.skip('Cannot test progress without qtpy.', allow_module_level=True)
+except RuntimeError:
+    pytest.skip(
+        'Cannot test progress without Qt bindings.', allow_module_level=True
+    )
+from ..progress import ProgressBar, progress
+
+
+def activity_dock_children(viewer):
+    return viewer.window.qt_viewer.activityDock.children()[4].children()
```

Comment:
```suggestion
from contextlib import contextmanager

def qt_viewer_has_pbar(qt_viewer):
    return bool(qt_viewer.activityDock.widget().findChild(ProgressBar))

@contextmanager
def assert_pbar_added_to(viewer):
    assert not qt_viewer_has_pbar(viewer.window.qt_viewer)
    yield
    assert qt_viewer_has_pbar(viewer.window.qt_viewer)
```

since this is a repeated pattern here to do something and then make sure it added a pbar, we can use a context manager to a) make sure one wasn't there before, and b) reduce duplication of the test